clipboard: macOS: fallback to tmux if pbcopy is broken

justinmk opened the pull request:
On some versions of macOS, pbcopy doesn't work in tmux <2.6
https://superuser.com/q/231130
Fallback to tmux in that case.

Add a healthcheck for this scenario.

mhinz commented on runtime/autoload/health/provider.vim:
```diff
@@ -120,6 +126,13 @@ endfunction
+    let tmux_version = matchstr(system('tmux -V'), '\d\+\.\d\+')
+    call health#report_error('pbcopy does not work with tmux version: '.tmux_version,
+          \ ['Install tmux 2.6+',
+          \  'See https://superuser.com/q/231130'])
```
Maybe we could add...

_or use tmux together with reattach-to-user-namespace: https://github.com/ChrisJohnsen/tmux-MacOSX-pasteboard_

...which is how people used the pasteboard before tmux 2.6.

justinmk committed:
clipboard: macOS: fallback to tmux if pbcopy is broken #7940

On some versions of macOS, pbcopy doesn't work in tmux <2.6
https://superuser.com/q/231130
Fallback to tmux in that case.

Add a healthcheck for this scenario.
